### DISCUSSION TITLE
Remove comment when starting new line

### DISCUSSION THREAD

**Discussion by kasshedi117** [numToStr/Comment.nvim#305] [Q&A] [answered]
Actually if I have a comment in Lua file for example

-- comment.setup [myCursor] {} 

from that line when I press 'O' to start typing in insert mode next line i got this

-- comment.setup {}
-- [myCursor]

instead I want to have 

-- comment.setup {}
[myCursor] 

without adding comment to the next line, thanks in advance.

**Comment by bew** [comment, marked as answer]
This is a builtin feature of nvim, you can control it by tweaking the `formatoptions` option, see [`:h formatoptions`](https://neovim.io/doc/user/options.html#'formatoptions') and [`:h fo-table`](https://neovim.io/doc/user/change.html#fo-table).

It's common that this option is overwritten by filetype runtime plugins, in which case you'll need to change it back the way you want.
For example for Lua, you can get your behavior like this:
```vim
" In ~/.config/nvim/after/ftplugin/lua.vim
" (to make sure it is run after $VIMRUNTIME/ftplugin/lua.vim which is loaded for every Lua file)
setlocal formatoptions-=o
```
> ```
> o   Automatically insert the current comment leader after hitting 'o' or
>     'O' in Normal mode.  In case comment is unwanted in a specific place
>     use CTRL-U to quickly delete it. [i_CTRL-U](https://neovim.io/doc/user/insert.html#i_CTRL-U)
> ```

**Reply by kasshedi117** [reply]
That fixed my issue, thanks for the quick reply.
